### PR TITLE
Run linkcheck on changed files only on PRs

### PR DESCRIPTION
**Pull request by TheTripleV**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #
 # You can set these variables from the command line.
 
-SPHINXOPTS    = -W --keep-going -q -T
+SPHINXOPTS    = -W --keep-going -T
```
Comment (Daltz333) [inline Makefile:5]: Is this change intended?

Comment (TheTripleV) [reply Makefile:5]: No, I'll undo it later. It's impossible for me to debug in quiet mode so I make that change. I accidentally commited it.

Comment (TheTripleV) [reply Makefile]: Ok undid that change

**Review by Daltz333 — APPROVED** [review]
